Purchases/FoundationExtensions/JSONDecoder+Extensions.swift

```diff
@@ -79,11 +79,16 @@ extension KeyedDecodingContainer {
     ///   - defaultValue: The default value returned if decoding fails.
     /// - Returns: A value of the requested type, or the given default value
     /// if decoding fails.
-    func decode<T: Decodable>(_ type: T.Type, forKey key: Self.Key, defaultValue: T) -> T {
+    func decode<T: Decodable>(_ type: T.Type,
+                              forKey key: Self.Key,
+                              defaultValue: T,
+                              logDecodingError: Bool = true) -> T {
```

Comment:
I mentioned it in the description, but maybe we want to not log anytime there's a `defaultValue` instead of adding a new parameter to this function.

Reply:
> maybe we want to not log anytime there's a defaultValue instead of adding a new parameter to this function.

That makes sense to me.

Reply:
It looks like the issue is about this polluting debug logs so that wouldn't really fix this issue, would it? 🤔  Do we have a verbose setting we could log this under?